### PR TITLE
Home page refactored

### PR DESCRIPTION
**Pull request by DimitriB01**
## Type of changes

- [ ] Bugfix 
- [ ] Feature 
- [x] Refactoring 
- [ ] Optimization 
- [ ] Documentation 
- [ ] CI/CD 
- [ ] Other 

## Requirements
- [x] Everything build out of Material UI components
- [x] New Material UI components capsulated via dependency injection in "@common/components"
- [x] Provide basic information pieces in comments
- [x] Establish interface to backend (if necessary) 
- [x] Use Skeleton components if something has to be fetched (if necessary)
- [x] Error management: Components should handle undefined inputs
- [x] Components should be resuable (if possible)
- [x] [Component Requirements](https://lab.las3.de/nextcloud/index.php/apps/onlyoffice/510074?filePath=%2FHASKI-Extern%2F06-Frontend%2F03-UX%2F04-Implementation%2FComponent_Requirements.docx) are met

### CODE REVIEW

**Comment by DimitriB01** [comment]
1 test schläft fehl, der wird in einem anderen Branch (Betrifft fetchen von Couse-Topics) gefixt

File: src/pages/Home/Home.test.tsx [hunk 1]
```diff
@@ -1,19 +1,94 @@
...
+  test('fetching User or Course throws error', async () => {
+    mockServices.getCourses.mockImplementation(() => {
+      throw new Error('Error')
+    })
+
+    jest.spyOn(console, 'error').mockImplementation(() => {
+      return
+    })
+
+    const addSnackbarMock = jest.fn()
```
Comment (theUpsider) [inline src/pages/Home/Home.test.tsx:69]: Mit dem mock machst du nix. Könnte man löschen bzw drauf testen.

File: src/pages/Home/Home.tsx [hunk 2]
```diff
@@ -7,11 +7,26 @@ import { AuthContext, SnackbarContext } from '@services'
 import { Stack } from '@mui/system'
 import { Card, CardContent, Skeleton, Typography } from '@mui/material'
 import { useNavigate } from 'react-router-dom'
```
Comment (theUpsider) [inline src/pages/Home/Home.tsx:9]: import { DefaultButton as Button, DefaultStack as Stack } from '@common/components' statt Mui import

**Review by theUpsider — CHANGES_REQUESTED** [review]
Kommentare anbei^^

File: src/common/core/Course/Course.ts [hunk 3]
```diff
@@ -1,3 +1,9 @@
+type CourseReturn = (userId?: number, lmsUserId?: number, studentId?: number) => Promise<CourseResponse>
+
+type CourseResponse = {
```
[the remark this answers is not in the document]
Comment (DimitriB01) [reply src/common/core/Course/Course.ts:3]: Man benötigt courses, da es die Struktur der Antwort darstellt:

{
    "courses": [
        {
            "created_at": null,
            "created_by": null,
            "id": 1,
            "last_updated": null,
            "lms_id": 2,
            "name": "Software Engineering",
            "university": "HS-RE"
        },
        {
            "created_at": null,
            "created_by": null,
            "id": 2,
            "last_updated": null,
            "lms_id": 2,
            "name": "Software Architecture",
            "university": "HS-AS"
        }
    ]
}

File: src/pages/Home/Home.tsx [hunk 4]
```diff
@@ -7,11 +7,26 @@ import { AuthContext, SnackbarContext } from '@services'
...
+import { Course } from '@core'
 
 /**
  *
  * @returns
  */
+
+//ToDo: create a Page for course id 99999 or make the button not clickable
+const initialCourses: Course[] = [
```
[the remark this answers is not in the document]
Comment (DimitriB01) [reply src/pages/Home/Home.tsx:18]: true

File: src/pages/Home/Home.tsx [hunk 5]
```diff
@@ -20,43 +35,43 @@ export const Home = () => {
...
-            message: error.message,
-            severity: 'error',
-            autoHideDuration: 5000
-          })
-        })
+            log.error(err.message)
+          }
+        }
+      }
     }
```
[the remark this answers is not in the document]
Comment (DimitriB01) [reply src/pages/Home/Home.tsx]: Hier braucht man 2 catch(), für jedes then jeweils. Aber sonst passt

File: src/common/services/Courses/getCourses.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { CourseReturn } from '@core'
```
Comment (theUpsider) [inline src/common/services/Courses/getCourses.ts:1]: CourseReturn und den anderen import noch zusammenfassen. Also dass beide aus @core sind ^^

**Review by theUpsider — CHANGES_REQUESTED** [review]
Nur one small change :D

**Pull request merged**